native/src/modules/categories/components/CategoryListContent.js

```diff
@@ -57,124 +53,41 @@ const CategoryListContent = ({
   const alterResources = useCallback(
     (node: HTMLNode) => {
       if (node.attribs) {
-        if (node.attribs.href) {
-          const newResource = cacheDictionary[decodeURI(node.attribs.href)]
-          if (newResource) {
-            node.attribs = {
-              ...(node.attribs || {}),
-              href: newResource
-            }
-            return node
-          }
-        } else if (node.attribs.src) {
-          const newResource = cacheDictionary[decodeURI(node.attribs.src)]
-          if (newResource) {
-            node.attribs = {
-              ...(node.attribs || {}),
-              src: newResource
-            }
-            return node
+        const newHref = node.attribs.href ?? cacheDictionary[decodeURI(node.attribs.href)]
+        const newSrc = node.attribs.src ?? cacheDictionary[decodeURI(node.attribs.src)]
+        if (newHref || newSrc) {
+          node.attribs = {
+            ...node.attribs,
+            ...newHref && { href: newHref },
+            ...newSrc && { src: newSrc },
```

Comment:
Are you sure this is doing what you expect it to do? It looks wrong to me, i.e. href is set if there was no href previously, same for src (at least afaict). I think the `??` operator is wrong, my suggestion would be (without having it tested):
```suggestion
        const newHref = node.attribs.href ? { href: cacheDictionary[decodeURI(node.attribs.href)] } : {}
        const newSrc = node.attribs.src ? { src: cacheDictionary[decodeURI(node.attribs.src)] }: {}
        node.attribs = {
          ...node.attribs,
          ...newHref,
          ...newSrc
```
Is there already a test that this works as expected?

Reply:
There is a test for navigating to a link. Anything else is not typically done by react-testing-library (such as checking html props)

Reply:
Still I had to replace the `??` with `&&` to make it work